Extensions/IISWebAppDeploy/Src/Tasks/IISWebAppDeploy/IISWebAppDeployV2/DeployIISWebApp.ps1

```diff
@@ -0,0 +1,113 @@
+﻿Import-Module $env:CURRENT_TASK_ROOTDIR\DeploymentSDK\InvokeRemoteDeployment.ps1
+
+Write-Verbose "Entering script DeployIISWebApp.ps1"
+
+function Trim-Inputs([ref]$package, [ref]$paramFile, [ref]$siteName, [ref]$adminUser)
+{
+    Write-Verbose "Triming inputs for excess spaces, double quotes"
+
+    $package.Value = $package.Value.Trim('"', ' ')
+    $paramFile.Value = $paramFile.Value.Trim('"', ' ')
+    $siteName.Value = $siteName.Value.Trim('"', ' ')
+
+    $adminUser.Value = $adminUser.Value.Trim()
+}
+
+function EscapeSpecialChars
+{
+    param(
+        [string]$str
+    )
+
+    return $str.Replace('`', '``').Replace('"', '`"').Replace('$', '`$')
+}
+
+function Get-ScriptToRun
+{
+    param (
+        [string]$webDeployPackage,
+        [string]$webDeployParamFile,
+        [string]$overRideParams,
+        [string]$websiteName,
+        [string]$removeAdditionalFiles,
+        [string]$excludeFilesFromAppData,
+        [string]$takeAppOffline,
+        [string]$additionalArguments
+    )
+
+    $msDeployScript = Get-Content  ./MsDeployOnTargetMachines.ps1 | Out-String
+    $overRideParams = EscapeSpecialChars -str $overRideParams
+    $websiteName = EscapeSpecialChars -str $websiteName
+    $AdditionalArguments = EscapeSpecialChars -str $AdditionalArguments
+
+    $invokeMain = "Execute-Main -WebDeployPackage `"$webDeployPackage`" -WebDeployParamFile `"$webDeployParamFile`" -OverRideParams `"$overRideParams`" -WebsiteName `"$websiteName`" -RemoveAdditionalFiles $removeAdditionalFiles -ExcludeFilesFromAppData $excludeFilesFromAppData -TakeAppOffline $takeAppOffline -AdditionalArguments `"$AdditionalArguments`""
+
+    Write-Verbose "Executing main function in MsDeployOnTargetMachines : $invokeMain"
+    $msDeployOnTargetMachinesScript = [string]::Format("{0} {1} ( {2} )", $msDeployScript,  [Environment]::NewLine,  $invokeMain)
+    return $msDeployOnTargetMachinesScript
+}
+
+function Run-RemoteDeployment
+{
+    param(
+        [string]$machinesList,
+        [string]$scriptToRun,
+        [string]$adminUserName,
+        [string]$adminPassword,
+        [string]$winrmProtocol,
+        [string]$testCertificate,
+        [string]$deployInParallel,
+        [string]$webDeployPackage
+    )
+
+    Write-Host "Starting deployment of IIS Web Deploy Package : $webDeployPackage"
```

Comment:
We can localize this string.

Reply:
Consider for all Write-host, write-output

Reply:
It is done